src/tools/sc-hsm-tool.c

```diff
@@ -1329,16 +1326,17 @@ static int wrap_key(sc_context_t *ctx, sc_card_t *card, int keyid, const char *o
 	sc_cardctl_sc_hsm_wrapped_key_t wrapped_key;
 	struct sc_pin_cmd_data data;
 	sc_path_t path;
+	sc_file_t *file = NULL;
 	FILE *out = NULL;
 	u8 fid[2];
-	u8 ef_prkd[MAX_PRKD];
-	u8 ef_cert[MAX_CERT];
+	u8 *ef_prkd = NULL;
+	u8 *ef_cert = NULL;
 	u8 wrapped_key_buff[MAX_KEY];
-	u8 keyblob[MAX_WRAPPED_KEY];
-	u8 *key;
-	u8 *ptr;
+	u8 *keyblob = NULL;
+	u8 *key = NULL;
+	u8 *ptr = NULL;
 	char *lpin = NULL;
-	size_t key_len;
+	size_t key_len, keyblob_len = MAX_KEY;
```

Comment:
If you initialize all the new pointers to `NULL`, I suggest you also initialize their length values to `0`